build.override.targets

```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="4.0"  xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <!--if ServiceUri is set, the service has been already started on another machine-->
+  <Target Name="WcfSetup" Condition="($(WithCategories.Contains('OuterLoop')) or '$(OuterLoop)' == 'true') and ('$(ServiceUri)' == '') and $(OS.Contains('Windows'))" BeforeTargets="BuildAllProjects">
```

Comment:
Keep in mind that what ends up happening as part of a traversal project here is that the BuildAllProjects target runs the default target on each child project and that default target is BuildAndTest so while it might be confusing the BuildAllProjects actually makes each project do a BuildAndTest individually, so it encapsulates the full build and test run. The BuildAndTest target in build.proj itself pretty much only does a build as you can see the Test target is empty https://github.com/dotnet/wcf/blob/master/dir.traversal.targets#L247. So what I think you essentially want is something like like:
```
    <TraversalBuildDependsOn>
      WcfSetup;
      $(TraversalBuildDependsOn);
      WcfCleanup;
    </TraversalBuildDependsOn>
```